RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -752,6 +761,18 @@ bool MuonIdProducer::isGoodRPCMuon( const reco::Muon& muon )
   return ( muon.numberOfMatchedRPCLayers( reco::Muon::RPCHitAndTrackArbitration ) > minNumberOfMatches_ );
 }
 
+bool MuonIdProducer::isGEMMuon( const reco::Muon& muon )
+{
+  if(muon.track()->pt() < minPt_ || muon.track()->p() < minP_) return false;
+  return ( muon.numberOfMatches( reco::Muon::GEMSegmentAndTrackArbitration ) >= 1 );    
+}
+
+bool MuonIdProducer::isME0Muon( const reco::Muon& muon )
+{
+  if(muon.track()->pt() < minPt_ || muon.track()->p() < minP_) return false;
```

Comment:
do you really want to apply the same pt cut for ME0 as used for CSC?
This value was selected/tuned to accept all muons within CSC acceptance.
ME0 goes beyond that.
pt of 0.5 GeV is probably low enough, for eta=3 this corresponds to p of 5 GeV. 
This is probably about 2 GeV above the rangeout.